e2e/suite1/connections-actions.test.ts

```diff
@@ -0,0 +1,295 @@
+        ],
+      },
+    ]);
+  });
+
+  it('should go back to the original entity and check the updated "Votos separados"', async () => {
+    await connections.clickEdit();
+    await connections.goToRelation(3, 1, 1);
+    await expect(page).toClick('#tab-connections');
+    const relations = await connections.readRelations();
+    expect(relations[1]).toEqual([
+      'Court',
+      {
+        'Votos separados': ['Anzualdo Castro'],
+      },
+    ]);
+  });
+
+  it('should remove "País" and "Votos separados", then undo the remove on "País"', async () => {
+    await connections.clickEdit();
+    await connections.clickRemoveTypeGroup(3, 2);
+    await connections.clickRemoveTypeGroup(2, 1);
+    await connections.clickRemoveTypeGroup(3, 2);
+    await connections.clickSave();
+    await connections.reloadPage();
+    const relations = await connections.readRelations();
+    expect(relations).toEqual([
+      [
+        'Court',
+        {
+          '': ['Acevedo Buendía y otros. Resolución de la CorteIDH de 28 de enero de 2015'],
+        },
+      ],
+      [
+        'Court',
+        {
+          País: ['Bolivia', 'Brazil', 'Colombia'],
+          Firmantes: ['Tracy Robinson', 'Máximo Cisneros Sánchez', 'Margarette May Macaulay'],
+        },
+      ],
+    ]);
+  });
+
+  it('should handle a larger set of actions all at once', async () => {
+    await connections.clickEdit();
+    await connections.removeHub(1);
+    await connections.moveRelationship(2, 1, 1, 2, 2);
+    await connections.clickRemoveRelationship(2, 1, 1);
+    await connections.clickRemoveRelationship(2, 2, 1);
+    await connections.clickRemoveRelationship(2, 2, 2);
+    await connections.addHub('Court');
+    await connections.addHub('Court');
+    await connections.addNewRelTypeToHub(3, 'Firmantes');
+    await connections.addNewRelTypeToHub(3, 'País');
+    await connections.addNewRelTypeToHub(4, 'País');
+    await connections.addNewConnection(3, 1, 'Manuel Banchi Gundián');
+    await connections.addNewConnection(3, 1, 'Marco Monroy Cabra');
+    await connections.addNewConnection(3, 2, 'Ecuador');
+    await connections.addNewConnection(3, 2, 'Guyana');
+    await connections.addNewConnection(3, 2, 'Paraguay');
+    await connections.addNewConnection(4, 1, 'Suriname');
+    await connections.addNewConnection(4, 1, 'Uruguay');
+    await connections.clickSave();
+    await connections.reloadPage();
+    const relations = await connections.readRelations();
+    expect(relations).toEqual([
+      [
+        'Court',
+        {
+          País: ['Tracy Robinson', 'Colombia'],
+          Firmantes: ['Margarette May Macaulay'],
+        },
+      ],
+      [
+        'Court',
+        {
+          Firmantes: ['Manuel Banchi Gundián', 'Marco Monroy Cabra'],
+          País: ['Ecuador', 'Guyana', 'Paraguay'],
+        },
+      ],
+      [
+        'Court',
+        {
+          País: ['Suriname', 'Uruguay'],
+        },
+      ],
```

Comment:
I really like this way of testing, it is immediately clear what the expected structure was.

Reply:
Credit goes to the original nightmare test, I've just mimicked it with puppeteer tools, since I've also thought it was a nice approach. :)